onebusaway-webapp/src/main/java/org/onebusaway/webapp/impl/TypedMessagesFactory.java

```diff
@@ -44,43 +46,35 @@ public void setMessagesClass(Class<?> messagesClass) {
 
   protected Object createInstance() throws IOException {
     String name = _messagesClass.getName();
-    name = "/" + name.replace('.', '/') + ".properties";
-    InputStream is = _messagesClass.getResourceAsStream(name);
-    if (is == null)
-      throw new IllegalStateException("Unable to find resources: " + name);
-    Properties p = new Properties();
-    p.load(is);
-
-    for (Method method : _messagesClass.getDeclaredMethods()) {
-      if (!p.containsKey(method.getName()))
-        _log.warn("missing message name: messagesType="
-            + _messagesClass.getName() + " property=" + method.getName());
-    }
-
-    Handler handler = new Handler(p);
+    name = "/" + name.replace('.', '/');  
+    ReloadableResourceBundleMessageSource messageSource = new ReloadableResourceBundleMessageSource();
+    messageSource.setBasename(name);
```

Comment:
Here we have the base name (eg: ArrivalAndDepartureMessages) and find the locale resources:
* ArrivalAndDepartureMessages_en.properties
* ArrivalAndDepartureMessages_es.properties
* ArrivalAndDepartureMessages_fr.properties
* ArrivalAndDepartureMessages_{lang}.properties

But on this method dont use the default ArrivalAndDepartureMessages.properties file.